docs/process/guidance/feature_flags/index.rst

```diff
@@ -0,0 +1,507 @@
+  specifying the concrete values (constraint_value) within a varying dimension
+  (constraint_setting), similar to the key/value map. For example, the
+  fundamental dimensions for processor systems are typically defined by the CPU
+  architecture (ISA on a high-level, but could be specialized to a
+  micro architecture) and the Operating System. Platforms in Bazel are designed
+  for configuration of host/build tools, e.g. select the appropriate compiler
+  when building for x86 vs. ARM.
+
+Since this guidance is meant to address feature configuration, we will only
+focus on **Build settings**
+
+For further details on the implementation of build settings, check out the
+`Bazel documentation <https://bazel.build/extending/config#user-defined-build-settings>`_.
+
+When a user is building/testing, the configuration is set via a combination of
+information passed from the user, via command line, and the default values
+configured in the build settings. Bazel also provides convenient indirections
+that abstract a collection of flags, namely
+`--config <https://bazel.build/docs/user-manual#config>`_.
+
+Bazel transitions
+^^^^^^^^^^^^^^^^^
+
+As aforementioned, configuration of a build/test is typically set by the user
+via command-line. **Transitions** break this pattern in order to support
+multi-configuration builds/tests in one go. This is accomplished by creating an
+entry point for the build that automatically configures its graph, ignoring the
+configuration set by the user.
+
+Transitions have drawbacks though:
+
+Let's imagine a scenario where we have a test depending on an application that
+depends on a library, represented by `Test -> Application -> Library`. Let's
+also assume that our flag/configuration affects the `Application` but not the
+`Library`. If one runs both versions of the test, with the feature
+enabled/disabled, one would assume the `Library` is reused and only the
+`Application` and `Test` would have to be re-built.
+
+Unfortunately, this is not the case, with transitions the entire build tree
+would be duplicated. This happens because Bazel has a core assumption for
+correctness: Isolation between different actions/configurations, i.e. no shared
+actions. A more detailed explanation can be found
+`here <https://github.com/bazelbuild/bazel/issues/14236#issuecomment-1332896717>`_.
+
+The Bazel maintainers and community have mitigated this issue, by accomplishing
+this de-duplication via the cache (but not on the build system itself!).
+Several efforts have been done over the years which finally converging in
+`Path Mapping <https://github.com/bazelbuild/bazel/discussions/22658>`_.
+It was confirmed using a Java example, which is the language with most advanced
+support, this is behaving as desired.
+
+Path mapping for C++ is `supported <https://github.com/bazelbuild/bazel/pull/22876>`_
+since Bazel version 7.3.0. Any internal custom rules in SCORE must also support this.
```

Comment:
>Any internal custom rules in SCORE must also support this.

What does this imply? Where is it documented?

Reply:
Basic google search for `bazel path mapping` brings up this:
https://github.com/bazelbuild/bazel/discussions/22658

Honestly, I don't want to put even more details regarding Path Mapping into the documentation.
When you want to write a custom rule you anyway need to read their documentation. This note here is just to make people aware that they need to take Path Mapping into account.